Tools/ardupilotwaf/boards.py

```diff
@@ -156,6 +156,9 @@ def configure_env(self, cfg, env):
             env.LINKFLAGS += [
                 '-Wl,--gc-sections',
             ]
+        if cfg.options.static:
+            # force static build
+            env.STATIC_LINKING = True
```

Comment:
I think this should be in an upper scope, like [here](https://github.com/ArduPilot/ardupilot/blob/master/wscript#L124). That way, we only check for `env.STATIC_LINKING` when necessary instead of looking at `cfg.options`.